ios/RNSScreen.mm

```diff
@@ -1058,50 +1058,32 @@ - (CGFloat)getCalculatedHeaderHeightIsModal:(BOOL)isModal
   return navbarHeight;
 }
 
-- (CGSize)getCalculatedStatusBarHeightIsModal:(BOOL)isModal
+- (CGFloat)getNavigationBarInsetIsModal:(BOOL)isModal
 {
 #if !TARGET_OS_TV
   BOOL isDraggableModal = isModal && ![self.screenView isFullscreenModal];
   BOOL isDraggableModalWithChildViewCtr =
       isDraggableModal && self.childViewControllers.count > 0 && self.childViewControllers[0] != nil;
 
-  // When modal is floating (we can grab its header), we don't want to calculate status bar in it.
-  // Thus, we return '0' as a height of status bar.
+  // When modal is floating (we can grab its header), we don't want to get inset of the navigation bar.
+  // Thus, we return '0' as an inset.
   if (isDraggableModalWithChildViewCtr || self.screenView.isTransparentModal) {
-    return CGSizeMake(0, 0);
+    return 0;
   }
 
-  CGSize fallbackStatusBarSize = [[UIApplication sharedApplication] statusBarFrame].size;
-
-#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && defined(__IPHONE_13_0) && \
-    __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_13_0
-  if (@available(iOS 13.0, *)) {
-    CGSize primaryStatusBarSize = self.view.window.windowScene.statusBarManager.statusBarFrame.size;
-    if (primaryStatusBarSize.height == 0 || primaryStatusBarSize.width == 0)
-      return fallbackStatusBarSize;
-
-    return primaryStatusBarSize;
-  } else {
-    return fallbackStatusBarSize;
-  }
-#endif /* Check for iOS 13.0 */
+  return self.navigationController.navigationBar.frame.origin.y;
 
 #else
-  // On TVOS, status bar doesn't exist
-  return CGSizeMake(0, 0);
+  // On TVOS there's no inset of navigation bar.
+  return 0;
 #endif // !TARGET_OS_TV
 }
 
 - (CGFloat)calculateHeaderHeightIsModal:(BOOL)isModal
 {
-  CGFloat navbarHeight = [self getCalculatedHeaderHeightIsModal:isModal];
-  CGSize statusBarSize = [self getCalculatedStatusBarHeightIsModal:isModal];
-
-  // Unfortunately, UIKit doesn't care about switching width and height options on screen rotation.
-  // We should check if user has rotated its screen, so we're choosing the minimum value between the
-  // width and height.
-  CGFloat statusBarHeight = MIN(statusBarSize.width, statusBarSize.height);
-  return navbarHeight + statusBarHeight;
+  CGFloat navbarHeight = [self getNavigationBarHeightIsModal:isModal];
+  CGFloat navbarInset = [self getNavigationBarInsetIsModal:isModal];
+  return navbarHeight + navbarInset;
 }
 
 - (void)calculateAndNotifyHeaderHeightChangeIsModal:(BOOL)isModal
```

Comment:
Just one comment from me - leaving the name `calculateHeaderHeightIsModal` is made on purpose here. For me `header` is the navigationBar + an inset of it - that's why I'm leaving it here.